### DISCUSSION TITLE
WARNING:urllib3.connectionpool:Connection pool is full, discarding connection: gitlab.naxions.com. Connection pool size: 10

### DISCUSSION THREAD

**Discussion by binbjz** [python-gitlab/python-gitlab#2301] [Q&A]
Hi,
  When I execute the program, it can get the correct results, but this warning indicates that the connection pool is full. How can I increase the size of the connection pool ?
  I modified urllib3/connectionpool.py - maxsize=1 to maxsize=20, but it didn't work. I also modified requests/adapters.py - DEFAULT_POOLSIZE = 10 to 20, but it didn't work too. Please help me to resolve this warning. Thanks！

python 3.11-rc2
python-gitlab         3.10.0

**Comment by nejch** [comment]
@binbjz this is more about the underlying urllib3 behavior so I'm going to link to upstream documentation. Generally I think you could just suppress/ignore them as well, as you'll find in some stackoverflow answers, but I'm also not sure about the specifics of your app.

https://urllib3.readthedocs.io/en/latest/advanced-usage.html#customizing-pool-behavior

To change this in requests:
https://stackoverflow.com/questions/18466079/change-the-connection-pool-size-for-pythons-requests-module-when-in-threading

You can provide a custom session to python-gitlab with these customized sessions:
https://python-gitlab.readthedocs.io/en/stable/api-usage-advanced.html#using-a-custom-session

**Comment by binbjz** [comment]
I will try it, thanks.
